homeassistant/components/tplink/entity.py

```diff
@@ -275,55 +257,94 @@ def _entities_for_device_and_its_children[_E: CoordinatedTPLinkEntity](
     *,
     feature_type: Feature.Type,
     entity_class: type[_E],
+    child_coordinators: list[TPLinkDataUpdateCoordinator] | None = None,
 ) -> list[_E]:
     """Create entities for device and its children.
 
     This is a helper that calls *_entities_for_device* for the device and its children.
     """
     entities: list[_E] = []
+    # Add parent entities before children so via_device id works.
+    entities.extend(
+        _entities_for_device(
+            device,
+            coordinator=coordinator,
+            feature_type=feature_type,
+            entity_class=entity_class,
+        )
+    )
     if device.children:
         _LOGGER.debug("Initializing device with %s children", len(device.children))
-        for child in device.children:
+        for idx, child in enumerate(device.children):
+            # HS300 does not like too many concurrent requests and its emeter data requires
+            # a request for each socket, so we receive separate coordinators.
+            if child_coordinators:
+                child_coordinator = child_coordinators[idx]
+            else:
+                child_coordinator = coordinator
             entities.extend(
                 _entities_for_device(
                     child,
-                    coordinator=coordinator,
+                    coordinator=child_coordinator,
                     feature_type=feature_type,
                     entity_class=entity_class,
                     parent=device,
                 )
             )
 
-    entities.extend(
-        _entities_for_device(
-            device,
-            coordinator=coordinator,
-            feature_type=feature_type,
-            entity_class=entity_class,
-        )
-    )
-
     return entities
 
 
+def _category_for_feature(feature: Feature | None) -> EntityCategory | None:
+    """Return entity category for a feature."""
```

Comment:
This should probably go back into the entity class?

Reply:
As we're moving `_description_for_feature` into the platform classes will leave this here so can be called from the platforms

Reply:
Does it need to? It seems ok as a static